addons/mail/sfu_server/src/services/http.js

```diff
@@ -0,0 +1,210 @@
+import os from "node:os";
+import http from "node:http";
+
+import * as ws from "#src/services/ws.js";
+import * as auth from "#src/services/auth.js";
+import * as config from "#src/config.js";
+import * as rtc from "#src/services/rtc.js";
+import { Logger, parseBody, extractRequestInfo } from "#src/utils/utils.js";
+import { SESSION_CLOSE_CODE } from "#src/models/session.js";
+import { Channel } from "#src/models/channel.js";
+
+/**
+ * @typedef {function} routeCallback
+ * @param {http.IncomingMessage} req
+ * @param {http.ServerResponse} res
+ * @param {Object} param2
+ * @param {string} param2.remoteAddress
+ * @param {string} param2.protocol
+ * @param {string} param2.host
+ * @return {http.ServerResponse}
+ */
+
+export const API_VERSION = 1;
+const logger = new Logger("HTTP");
+let httpServer;
+
+/**
+ * @param {Object} [options]
+ * @param {string} [options.httpInterface]
+ * @param {number} [options.port]
+ */
+export async function start({ httpInterface = config.HTTP_INTERFACE, port = config.PORT } = {}) {
+    logger.info("starting...");
+    const routeListener = new RouteListener();
+    routeListener.get(`/v${API_VERSION}/noop`, {
+        callback: (req, res) => {
+            res.statusCode = 200;
+            return res.end(JSON.stringify({ result: "ok" }));
+        },
+    });
+    routeListener.get(`/v${API_VERSION}/stats`, {
+        callback: async (req, res, { remoteAddress }) => {
+            res.setHeader("Content-Type", "application/json");
+            req.statusCode = 200;
+            const heapTotal = process.memoryUsage().heapTotal;
+            const loadAvg = os.loadavg();
+            logger.info(
+                `[${remoteAddress}] fetched stats: last minute load average: ${loadAvg[0]}, total memory: ${heapTotal}`
+            );
+            return res.end(
+                JSON.stringify({
+                    workerStats: await rtc.getWorkersResourceUsages(),
+                    heapTotal,
+                    loadAvg,
```

Comment:
I'd rather have the content of `SIGIO` and `SIGPWR` availble as JSON. This will provide more valuable info (number of channels and sessions) which can be tracked over time. The number of sessions per channel is also useful (= participant per channel ?) is also useful.